### PR TITLE
feat: introduce bigtable-hbase-shaded universal artifact

### PR DESCRIPTION
**Pull request by igorbernstein2**
This artifact contains support for both HBase 1 and HBase 2. This will be the primary artifact that endusers will be use To help in the transition the old -shaded artifacts will simply depend on this new artifact

Change-Id: I1449e42b7dbd51353110f6c27ab74ddea963441f

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml [hunk 1]
```diff
@@ -62,20 +66,20 @@ limitations under the License.
...
-        in bigtable-hbase-1.x-shaded/pom.xml -->
+        in bigtable-hbase-shaded/pom.xml -->
         <exclusion>
           <groupId>${project.groupId}</groupId>
           <artifactId>bigtable-hbase-1.x</artifactId>
         </exclusion>
         <exclusion>
-          <groupId>com.google.cloud.bigtable</groupId>
-          <artifactId>bigtable-metrics-api</artifactId>
+          <groupId>${project.groupId}</groupId>
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml:81]: revert to hard coded group id

File: hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/pom.xml [hunk 2]
```diff
@@ -58,18 +58,22 @@ limitations under the License.
...
       <version>2.7.3-SNAPSHOT</version> <!-- {x-version-update:bigtable-client-parent:current} -->
 
       <!-- Workaround MNG-5899 & MSHADE-206. Maven >= 3.3.0 doesn't use the dependency reduced
       pom.xml files when invoking the build from a parent project. So we have to manually exclude
       the dependencies. Note that this works in conjunction with the manually promoted dependencies
-      in bigtable-hbase-1.x-shaded/pom.xml -->
+      in bigtable-hbase-shaded/pom.xml -->
       <exclusions>
         <exclusion>
           <groupId>${project.groupId}</groupId>
```
Comment (kolea2) [inline hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/pom.xml:70]: same here and below

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java [hunk 3]
```diff
@@ -43,54 +43,54 @@ public class BigtableConfiguration {
...
-            + "failed to find bigtable-hbase-1.x on the classpath.");
-    return CONNECTION_CLASS;
-  }
+    String hbaseVersion = VersionInfo.getVersion();
 
-  private static Class<?> getConnectionRegistryClass() {
+    final String connectionClassName;
+
+    if (VersionInfo.compareVersion(hbaseVersion, "2.0") >= 0) {
+      connectionClassName = "com.google.cloud.bigtable.hbase2_x.BigtableConnection";
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java:59]: please add some tests for this

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java (hunk 3 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java:59]: This is already covered by the existing ConnectionTests:
https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableConnection.java

https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableConnection.java

File: bigtable-client-core-parent/bigtable-hbase-shaded/pom.xml [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    <artifactId>bigtable-client-core-parent</artifactId>
+    <groupId>com.google.cloud.bigtable</groupId>
+    <version>2.7.3-SNAPSHOT</version> <!-- {x-version-update:bigtable-client-parent:current} -->
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>bigtable-hbase-shaded</artifactId>
+
+  <properties>
+    <maven.compiler.source>11</maven.compiler.source>
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase-shaded/pom.xml:15]: eek, good catch

File: bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml (hunk 1 above)
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml]: done

**Review by kolea2 — APPROVED** [review]

**Pull request merged**